PasswordVault.Services/EncryptionService/AesEncryption.cs

```diff
@@ -56,58 +56,30 @@ public AesEncryption(EncryptionParameters parameters)
-                (byte)_encryptionParameters.KeyDerivationParameters.MemorySizeKb,
-                (byte)(_encryptionParameters.KeyDerivationParameters.MemorySizeKb >> 8),
-                (byte)(_encryptionParameters.KeyDerivationParameters.MemorySizeKb >> 16),
-                (byte)(_encryptionParameters.KeyDerivationParameters.MemorySizeKb >> 24),
-                (byte)_encryptionParameters.KeyDerivationParameters.DegreeOfParallelism,
-                (byte)_encryptionParameters.Mac, 
-                
-            };
+            byte[] cipherSuite = EncryptionHelpers.PackCipherParamsIntoBytes(_encryptionParameters);
             byte[] authenticateHash;
             byte[] cipherBytes;
             byte[] cipherkey;
             byte[] hmackey;
-
-            var keysize = _encryptionParameters.KeyDerivationParameters.KeySizeBytes;
+            var keysize = _encryptionParameters.KeyDerivationParameters.KeySizeBytes; // dont allow configurable key size for AES
             var hmacsize = _integrityVerification.GetHMACKeySizeInBits(_encryptionParameters.Mac).ToNumBytes();
+
+            // Derive keys for AES and HMAC
             IKeyDerivation keyDerivation = KeyDerivationFactory.Get(_encryptionParameters.KeyDerivationParameters.Algorithm);
-            var combinedKey = keyDerivation.DeriveKey(passPhrase, salt, _encryptionParameters.KeyDerivationParameters, keysize + hmacsize); // hash a key for both aes and hmac
+            var combinedKey = keyDerivation.DeriveKey(passPhrase, salt, _encryptionParameters.KeyDerivationParameters, keysize + hmacsize);
             cipherkey = combinedKey.Take(keysize).ToArray();
             hmackey = combinedKey.Skip(keysize).Take(hmacsize).ToArray();
             
-            using (SymmetricAlgorithm cipher = Aes.Create())
+            // Encrypt data using AES
+            using (SymmetricAlgorithm cipher = Aes.Create()) // Use Aes.Create to get the best implementation of AES
             {      
-                cipher.BlockSize = _encryptionParameters.BlockSizeBytes.ToNumBits();
-
-                switch(_encryptionParameters.Algorithm)
-                {
-                    case EncryptionAlgorithm.Aes128CfbPkcs7:
-                    case EncryptionAlgorithm.Aes256CfbPkcs7:
-                        cipher.Mode = CipherMode.CFB;
-                        cipher.Padding = PaddingMode.PKCS7;
-                        break;
-                    case EncryptionAlgorithm.Rijndael128CbcPkcs7:
-                    case EncryptionAlgorithm.Rijndael256CbcPkcs7:
-                        cipher.Mode = CipherMode.CBC;
-                        cipher.Padding = PaddingMode.PKCS7;
-                        break;
-                    case EncryptionAlgorithm.Unknown:
-                    default:
-                        throw new Exception();
-                }
+                cipher.BlockSize = _encryptionParameters.BlockSizeInBytes.ToNumBits();
```

Comment:
This should always be 128bits for AES, either add sanity check or hardcode to ignore settings.

Reply:
Blocksize parameters already throws exception.